tests/functional/test_snap.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python3
+"""Test exporter snap."""
+import os
+import re
+from subprocess import run
+
+SNAP_NAME = "prometheus-juju-backup-all-exporter"
+
+
+def test_default_config_installed(snap_config):
+    """Check if the snap default config exists."""
+    assert os.path.exists(snap_config)
+
+
+def test_snap_active(snap_name):
+    """Check if the snap is in active state."""
+    result = run(
+        f"systemctl is-active snap.{snap_name}.{snap_name}.service",
+        shell=True,
+        capture_output=True,
+    )
+    assert result.returncode == 0
+    assert result.stdout.decode().strip() == "active"
+
+
+def test_exporter_http_server():
+    """Check if http server is running."""
+    result = run("curl http://localhost:10000", shell=True, capture_output=True)
+    assert result.returncode == 0
+
+
+def test_exporter_metrics_created(available_metric_names):
+    """Check if juju-backup-all related metric is created."""
+    result = run("curl http://localhost:10000", shell=True, capture_output=True)
+    assert result.returncode == 0
+
+    output = result.stdout.decode().strip()
+    for metric_name in available_metric_names:
+        pattern = metric_name
+        match = re.search(pattern, output).group()
+        assert match is not None
```

Comment:
Thanks for the comment. This has been taken care in the `conftest.py`, I used `sleep(5)` to handle this, and it works pretty well for this small exporter. In general, I agree that we may use retry to wait util the exporter fully functioning.

Reply:
`sleep` is acceptable, but it would be much better to use retry for better practices. But again, this is not a deal-breaker IMO

Reply:
I've change it to use retry with timeout instead of pure sleep to wait for the exporter until it's up and ready for requests. This will also address all the `curl` commands in the test cases that failed with `Connection Refused` because the exporter will at least be queried once (so that metrics will be created) before any test case .